### DISCUSSION TITLE
Several requests: See IP:port on alert, new columns, filter log view, DPI scaling clarity, etc

### DISCUSSION THREAD

**Discussion by kstrikis** [tnodir/fort#358] [Ideas]
I am excited to have learned about Fort, and hope it can be the best Windows firewall choice. I did not want to dump all my ideas into new issues right away in case that is not courteous. I have looked through the open issues and saw that Allowed Connections, profiles, and domain name filtering are already opened.

- [x] feature so on alert, see not just program that initiated the connection request, but the connection's local IP, local port, remote IP, remote port, protocol, and direction inbound/outbound (like Windows Firewall Control's notification window)

- [x] also maybe see related rules? so if there is an existing rule for that program or service, you could e.g. edit existing rule to add this IP/port

- [x] feature to add/hide new columns for separate IP and port in Statistics/Blocked Connections, not just IP:Port columns (and hopefully the same for allowed connections in the future)

- [ ] feature to sort columns in Statistics/Blocked connections (and hopefully the same for allowed connections in the future)

- [ ] feature to optionally filter out certain programs (or addresses, or ports, or anything that's local-only, etc.) from view in Statistics/Block Connections (and hopefully the same for allowed connections in the future)

- [x] feature to create wildcard rule from within alert

- [x] feature to see which Program listing, Rule, Zone, Group, and/or Schedule triggered a particular connection to be blocked in Statistics/Blocked Connections (and hopefully the same for allowed connections in the future)

- [ ] feature to see which program initiated DNS lookup?

- [ ] fix text clarity when DPI scaling is at e.g. 150%
  Picture below. Circled in red look somewhat wrong. Circled in green look okay to me.
<img width="998" alt="fort-text-issue" src="https://github.com/user-attachments/assets/1bf5cc3f-3280-4ed7-8e67-1a021d45c0c5">

**Comment by tnodir** [comment]
> fix text clarity when DPI scaling is at e.g. 150%

Try to:
- rename the `qt.conf` file to `old-qt.conf` in the installation folder
- restart Fort Firewall

The `qt.conf` forces to use old "GDI" font engine by Qt.

New "DirectWrite" font engine caches all fonts in memory and consumes e.g. 30MB against 15MB with old one for me: [QTBUG-129933](https://bugreports.qt.io/browse/QTBUG-129933).

**Comment by tnodir** [comment]
> feature so on alert, see not just program that initiated the connection request, but the connection's local IP, local port, remote IP, remote port, protocol, and direction inbound/outbound (like Windows Firewall Control's notification window)

It will be added after implementing an "Ask to connect" filter mode. So you can view paused connections.

**Comment by tnodir** [comment]
> feature to create wildcard rule from within alert

Implemented in [v3.15.3](https://github.com/tnodir/fort/releases/tag/v3.15.3).

**Comment by tnodir** [comment]
> feature to add/hide new columns for separate IP and port in Statistics/Blocked Connections, not just IP:Port columns (and hopefully the same for allowed connections in the future)

Implemented in [v3.16.4](https://github.com/tnodir/fort/releases/tag/v3.16.4).

**Comment by tnodir** [comment]
> feature so on alert, see not just program that initiated the connection request, but the connection's local IP, local port, remote IP, remote port, protocol, and direction inbound/outbound (like Windows Firewall Control's notification window)

Implemented in [v3.16.6](https://github.com/tnodir/fort/releases/tag/v3.16.6).
